controllers/controller/workspacerouting/workspacerouting_controller.go

```diff
@@ -71,23 +73,31 @@ func (r *WorkspaceRoutingReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 		return reconcile.Result{}, err
 	}
 
+	if instance.Annotations[config.WorkspaceExternalRoutingAnnotation] != "" {
```

Comment:
If the user for some reason(occasionally) annotates the embedded routing with it, then such CR is going to hang on removal since finalizers won't be handled.

Reply:
In general we'll have a tricky situation in switching between a supported routingclass and an external one -- if we initially set e.g. `basic`, and then switch to `che-workspace`, the controller can't know to remove old routing objects unless it also risks removing routing objects created by the external controller. I'll look into it more.

Should we make routingclass immutable once a workspace is created? I think it might make things cleaner here.

Reply:
> Should we make routingclass immutable once a workspace is created? I think it might make things cleaner here.

Sounds good to me

Reply:
Not sure I'm a bit fan of this limitation, unless it cannot be avoided

Reply:
Can't this be fixed by what @sleshchenko proposed in comment https://github.com/devfile/devworkspace-operator/pull/216#discussion_r535979658 ?
If in the status you add the `routingClass` that was used during the last reconciliation, then that means that each controller would also be able to read this value and perform any required cleaning, no ?